packages/web-components/src/badge/badge.ts

```diff
@@ -19,6 +27,20 @@ export class Badge extends FASTElement {
   @attr
   public appearance: BadgeAppearance = BadgeAppearance.filled;
 
+  /**
+   * Handles changes to appearance attribute custom states
+   * @param prev - the previous state
+   * @param next - the next state
+   */
+  public appearanceChanged(prev: BadgeAppearance | undefined, next: BadgeAppearance | undefined) {
+    if (prev) {
+      toggleState(this.elementInternals, `${prev}`, false);
+    }
+    if (next) {
+      toggleState(this.elementInternals, `${next}`, true);
+    }
+  }
```

Comment:
It would be nice if the `attr` decorator could have an option to associate attribution with the component's custom state, so that it can automatically add state toggle functionalities.

Reply:
We'd need to consider this in the context of FAST where this originates - given we want to stay closely aligned with the platform, I'm not sure how feasible it would be to upstream that and couple these...though we could look down the road at having our own `attr` decorator which does this.